pyteal/ast/router.py

```diff
@@ -459,34 +560,92 @@ def wrap_handler(
                     abi.MethodReturn(output_temp),
                     Approve(),
                 )
+        else:
+            subroutine_ify = ASTBuilder.__de_abify_subroutine(handler)
+            return Seq(
+                subroutine_ify(),
+                Approve(),
+            )
+
+    @staticmethod
+    def __de_abify_subroutine(
+        handler: ABIReturnSubroutine | SubroutineFnWrapper | Expr,
+    ) -> SubroutineFnWrapper:
+        handler = ASTBuilder.__handler_validity_check(handler)
+        (
+            arg_vals,
+            app_arg_vals,
+            txn_arg_vals,
+        ) = ASTBuilder.__subroutine_argument_instance_generate(handler)
+
+        (
+            decode_instructions,
+            arg_vals,
+            proto,
+        ) = ASTBuilder.__decode_constructions_and_args(
+            arg_vals,
+            app_arg_vals,
+            txn_arg_vals,
+            handler,
+            use_frame_pt=True,
+        )
+
+        subroutine_caster = Subroutine(TealType.none, f"{handler.name()}_caster")
+        return_expr: Expr
+
+        if not proto or not proto.mem_layout:
+            raise TealInternalError(
+                "Assuming to be using frame pointer, memory layout and proto should be available."
+            )
+
+        if handler.type_of() == sdk_abi.Returns.VOID:
+            return_expr = Seq(
+                *proto.mem_layout._succinct_repr(),
+                *decode_instructions,
+                cast(Expr, handler(*arg_vals)),
+            )
+        else:
+            output_temp: abi.BaseType = cast(
+                OutputKwArgInfo, handler.output_kwarg_info
+            ).abi_type.new_instance()
+            output_temp._stored_value = FrameVar(proto, 0)
+            subroutine_call: abi.ReturnedValue = cast(
+                abi.ReturnedValue, handler(*arg_vals)
+            )
+
+            return_expr = Seq(
+                *proto.mem_layout._succinct_repr(),
+                *decode_instructions,
+                subroutine_call.store_into(output_temp),
+                abi.MethodReturn(output_temp),
+            )
```

Comment:
```suggestion
        decoding = [*proto.mem_layout._succinct_repr(), *decode_instructions]
        if handler.type_of() == sdk_abi.Returns.VOID:
            returning = cast(Expr, handler(*arg_vals))
        else:
            output_temp: abi.BaseType = cast(
                OutputKwArgInfo, handler.output_kwarg_info
            ).abi_type.new_instance()
            output_temp._stored_value = FrameVar(proto, 0)
            subroutine_call: abi.ReturnedValue = cast(
                abi.ReturnedValue, handler(*arg_vals)
            )
            returning = [subroutine_call.store_into(output_temp), abi.MethodReturn(output_temp)]
        return Seq(*decoding, *returning)
```
please ignore this suggestion if you disagree

Reply:
I am taking with some minor changes.